src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -355,10 +356,41 @@ static QString AnalyzeURI( QString const &uri,
 
       else if ( field == QLatin1String( "geometrytype" ) )
       {
-        ogrGeometryTypeFilter = ogrWkbGeometryTypeFromName( value );
+        ogrGeometryTypeFilter = QgsOgrProviderUtils::wkbGeometryTypeFromName( value );
+      }
+      else if ( field == QLatin1String( "geometryname" ) )
+      {
+        geometryName = value;
+      }
+      else if ( field == QLatin1String( "geometryid" ) )
+      {
+        geometryIndex = value.toInt( &ok );
+      }
+      else if ( field == QLatin1String( "featurescounted" ) )
+      {
+        if ( value != QLatin1String( "Unknown" ) )
+        {
+          featuresCounted = value.toInt( &ok );
+        }
+      }
+      else if ( field == QLatin1String( "ogrtype" ) )
+      {
+        ogrType = value.toInt( &ok );
       }
     }
-
+    if ( geometryName == QString::null )
+    {
+      if ( ( layerName.contains( "(" ) ) && ( layerName.endsWith( QString( ")" ) ) ) )
```

Comment:
I think supporting layer_name(geometry_name) which only worked for SQLite/SPatialite is really a marginal use case, that hardly anybody knew about, and I wouldn't mind breaking it in QGIS 3. The less code in this PR, the more likely it will be merged.